### PR TITLE
Queries with transactions and aggregator functions

### PR DESCRIPTION
**Pull request by sumon766**
- Update the species in animals table
Digimon   Pokemon transaction

- Delete all records from animals table and then rollback and verify if data exists
Delete   Rollback from animals table

- Delete all records of animals born after 01.01.2022, create savepoint, update animals weight by multiplying -1, rollback to savepoint, update all negative weights from animals table and commit changes.
DELETE-SAVEPOINT-UPDATEWEIGHT-COMMIT

- Queries for answering questions using aggregator function
Last queries 1-3
Last queries 4-finish

### CODE REVIEW

File: schema.sql [hunk 1]
```diff
@@ -10,3 +11,16 @@ CREATE TABLE animals (
...
+
+/*Day 2 queries*/
+CREATE TABLE animals (
+    id INT PRIMARY KEY GENERATED ALWAYS AS IDENTITY,
+    name VARCHAR(100),
+    date_of_birth DATE,
+    escapte_attempts INT,
+    neutered BOOLEAN,
+    weight_kg DECIMAL(10,2),
+    species VARCHAR(100)
```
Comment (Davidosky007) [inline schema.sql:24]: * Kindly add the species column the right way below 👇 is an example of how to 
```
ALTER TABLE table_name
ADD column_name datatype;
```

File: data.sql [hunk 2]
```diff
@@ -1,6 +1,18 @@
...
 
+/*Day 1 data*/
 INSERT INTO animals (name, date_of_birth, escapte_attempts, neutered, weight_kg) VALUES ('Agumon', '2020-02-03', 0, true, 10.23);
 INSERT INTO animals (name, date_of_birth, escapte_attempts, neutered, weight_kg) VALUES ('Gabumon', '2018-11-15', 2, true, 8);
 INSERT INTO animals (name, date_of_birth, escapte_attempts, neutered, weight_kg) VALUES ('Pikachu', '2021-01-07', 1, false, 15.04);
 INSERT INTO animals (name, date_of_birth, escapte_attempts, neutered, weight_kg) VALUES ('Devimon', '2017-05-12', 5, true, 11);
+/*Day 1 data*/
+
+/*Day 2 data*/
+INSERT INTO animals (name, date_of_birth, escapte_attempts, neutered, weight_kg) VALUES ('Charmander', '2020-02-08', 0, false, 11);
```
Comment (Davidosky007) [inline data.sql:11]: * Kindly add the right weight value here which is (-11) as [required](https://github.com/microverseinc/curriculum-databases/blob/main/db-structure/vet_clinic_query_update_tables.md#:~:text=Animal%3A%20His%20name%20is%20Charmander.%20He%20was%20born%20on%20Feb%208th%2C%202020%2C%20and%20currently%20weighs%20%2D11kg.%20He%20is%20not%20neutered%20and%20he%20has%20never%20tried%20to%20escape.)

File: queries.sql [hunk 3]
```diff
@@ -1,10 +1,46 @@
...
 SELECT * FROM animals WHERE name != 'Gabumon';
-SELECT * FROM animals WHERE weight_kg BETWEEN 10.4 AND 17.3;
\ No newline at end of file
+/*Day 1 queries*/
+
+-- Day 2 queries
+BEGIN;
+UPDATE animals SET species = 'unspecified';
+SELECT * FROM animals;
+ROLLBACK;
```
Comment (Davidosky007) [inline queries.sql:17]: * Kindly verify that change was undone as [required ](https://github.com/microverseinc/curriculum-databases/blob/main/db-structure/vet_clinic_query_update_tables.md#:~:text=Inside%20a%20transaction%20update%20the%20animals%20table%20by%20setting%20the%20species%20column%20to%20unspecified.%20Verify%20that%20change%20was%20made.%20Then%20roll%20back%20the%20change%20and%20verify%20that%20the%20species%20columns%20went%20back%20to%20the%20state%20before%20the%20transaction.)

**Review by Davidosky007 — CHANGES_REQUESTED** [review]
Hi @sumon766 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
* Most data added properly 👍 
* Most Queries written properly 👍 
* Good looking readme 👍 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @davidsoky007 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: queries.sql (hunk 3 above)
Comment (sumon766) [reply queries.sql:17]: Verify that unspecified was undone

File: schema.sql (hunk 1 above)
Comment (sumon766) [reply schema.sql:24]: DROP and ADD species column again

File: data.sql (hunk 2 above)
Comment (binhussen) [inline data.sql:11]: - Great Job for implementing most of the previous reviewer suggestions, I think you missed one thing which is updating Charmander's weight to -11kg, so please update to fit the [requirment](https://github.com/microverseinc/curriculum-databases/blob/main/db-structure/vet_clinic_query_update_tables.md#:~:text=Animal%3A%20His%20name%20is%20Charmander.%20He%20was%20born%20on%20Feb%208th%2C%202020%2C%20and%20currently%20weighs%20%2D11kg.%20He%20is%20not%20neutered%20and%20he%20has%20never%20tried%20to%20escape) 👍

**Review by binhussen — CHANGES_REQUESTED** [review]
Hi @sumon766 , 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't been addressed yet.


## Highlights
- Well Structured Query✔️
- Descriptive pull request with Screenshoot✔️
- Used Git flow ✔️


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: data.sql (hunk 2 above)
Comment (sumon766) [reply data.sql:11]: @binhussen Didn't you see that I have added the following query in my queries.sql file?

UPDATE animals SET weight_kg = -11 WHERE id = 1;

**Review by Nemwel-Boniface — COMMENTED** [review]
## Hello @sumon766  👋,

I am Nemwel your reviewer.


## _This is an additional review as a result of the second opinion request._
I have received your second opinion request complaining about the last reviewer @binhussen 

While we acknowledge that there are many ways of implementing things and your implementation is still very correct. I believe the reviewer might have missed this and that is why he requested the changes. However, it would have been easier to instead change it "manually".

For this, I believe your project can be approved :rocket: 

Kindly re-submit your project in the dashboard and the next reviewer will approve it automatically. :rocket: 

- [ ] [OPTIONAL] It would be great to resolve the merge conflicts that you currently have before merging the project.

In case anything is not 100% clear tag me @Nemwel-Boniface so that I get the notification.

Keep rocking 🚀!

**Review by danmainah — APPROVED** [review]
Hi @sumon766 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🎉 🎉 

## Highlights

✔️ Linter setup correctly and all checks are passing
✔️ All queries and data added correctly

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**